amftrack/pipeline/functions/image_processing/experiment_class_surf.py

```diff
@@ -700,38 +739,42 @@ def is_in(self, t: int) -> bool:
         """Determines if the node is present at timestep t in the graph."""
         return self.label in self.experiment.nx_graph[t].nodes
 
-    def degree(self, t):
+    def degree(self, t) -> int:
+        """Return the degree of the node."""
         return self.experiment.nx_graph[t].degree(self.label)
 
-    def edges(self, t):
+    def edges(self, t) -> List["Edge"]:
+        """Return all the edges connected to this node"""
         return [
             self.experiment.get_edge(self, neighbour)
             for neighbour in self.neighbours(t)
         ]
 
-    def pos(self, t):
+    def pos(self, t) -> coord_int:
+        "Return coordinates in the general referential at timestep t"
         return self.experiment.positions[t][self.label]
 
     def ts(self) -> List[int]:
         """Returns a list of all the timestep at which the node is present."""
         return [t for t in range(len(self.experiment.nx_graph)) if self.is_in(t)]
 
-    def show_source_image(self, t: int, tp1: int):
+    def show_source_image(self, t: int, tp1=None):
         """
         Two use cases:
-        If t==tp1:
+        If tp1 is not provided:
         This function plots the original image from time t containing the node.
         As there can be several original images (because of overlap), the
-        darker image is choosen (why?)

```

Comment:
So, when you want to plot two timesteps on top of each other, nothing really guarantees that
1. The two pictures are in the same referential
2. The picture selected for each timestep is "the same"
So this is not meant as a perfect function, it was really developped early on to quickly check the quality of tracking and potentially troublecheck it.
The best function to use is actually probably `plot_raw_plus` where you can select a radius around the nodes of interest. Further improvement of it should be to give the possibility to plot in full quality. One however has to be conscious that the stitched image is somewhat transformed (background removal) so you cannot entirely trust what you see.

Reply:
And yes, darkest is chosen to maximise the chance that the image chosen in the two timesteps is "the same".

Reply:
Does it work choosing the darkest?
Wouldn't it be simpler to always take the one the more to the right and to the top for example? As when a position is in two images, it is always on the right of one and on the left of the other, or on the top of one and on the bottom of the other.

Reply:
Yes it's a bit of a stupid choice. I probably didn't want to think that day.